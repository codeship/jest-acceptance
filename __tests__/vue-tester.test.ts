import Vue from 'vue'
import VueTester from '../lib/vue-tester'
import { mount } from '@vue/test-utils'

const TestComponent = Vue.extend({
  template: `
    <div>
      <p>{{ message }}</p>
      <input name="message" type="text" v-model="inputValue"/>
      <button @click="buttonClick">Click Me!</button>
      <ul class="list">
        <li class="old-message">Hello Test <button @click="buttonClick">Remove</button></button></li>
      </ul>
    </div>
  `,

  data () {
    return {
      inputValue: '',
      message: 'Hello World'
    }
  },

  methods: {
    buttonClick () {
      this.message = this.inputValue
    }
  }
})

describe('VueTester', () => {
  describe('#fillIn', () => {
    it('updates the named input value', () => {
      const wrapper = mount(TestComponent)
      const tester = new VueTester(wrapper)

      tester.fillIn('message', 'Hello Universe!')
      expect(wrapper.vm.inputValue).toBe('Hello Universe!')
    })
  })

  describe('#click', () => {
    it('it triggers a click on the selected element', () => {
      const buttonClick = jest.fn()
      const wrapper = mount(TestComponent, {
        methods: { buttonClick }
      })
      const tester = new VueTester(wrapper)

      tester.click('button')
      expect(buttonClick).toHaveBeenCalled()
    })

    it('allows for deep selectors', () => {
      const buttonClick = jest.fn()
      const wrapper = mount(TestComponent, {
        methods: { buttonClick }
      })

      const tester = new VueTester(wrapper)

      tester.click('.list .old-message button')
      expect(buttonClick).toHaveBeenCalled()
    })
  })

  describe('#integration', () => {
    it('produces expected diffs', () => {
      const wrapper = mount(TestComponent)
      const tester = new VueTester(wrapper)

      expect(tester.current()).toMatchSnapshot()

      tester.fillIn('message', 'Hello Universe!')
      tester.click('button')
      expect(tester.next()).toMatchSnapshot()
    })
  })
})
