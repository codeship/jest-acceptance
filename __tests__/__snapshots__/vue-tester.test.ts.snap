// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VueTester #integration produces expected diffs 1`] = `
<div>
  <p>Hello World</p>
  <input name="message" type="text">
  <button>Click Me!</button>
</div>
`;

exports[`VueTester #integration produces expected diffs 2`] = `
- Diff A
+ Diff B

  <div>
-   <p>Hello World</p>
+   <p>Hello Universe!</p>
    <input name="message" type="text">
    <button>Click Me!</button>
  </div>
`;
